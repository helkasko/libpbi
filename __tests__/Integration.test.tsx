import { mount } from 'enzyme';
import * as React from 'react';
import {
  EmbeddedReport,
  EmbedTokenReportConfig,
  EmbedTokenVisualConfig,
  ReportWithEmbedToken,
  VisualWithEmbedToken
} from '../src';

describe('Library', () => {
  test('User can render a report with embed token', () => {
    const config: EmbedTokenReportConfig = {
      accessToken: 'token',
      groupId: 'groupId',
      reportId: 'reportId',
      description: 'description'
    };

    const report: ReportWithEmbedToken = new ReportWithEmbedToken(config);

    const embeddedReport = mount(<EmbeddedReport report={report} />);

    expect(embeddedReport).toMatchSnapshot();
  });

  test('User can render a visual with embed token', () => {
    const config: EmbedTokenVisualConfig = {
      accessToken: 'token',
      groupId: 'groupId',
      reportId: 'reportId',
      description: 'description',
      pageName: 'pageName',
      visualName: 'visualName',
    };

    const report: VisualWithEmbedToken = new VisualWithEmbedToken(config);

    const embeddedReport = mount(<EmbeddedReport report={report} />);

    expect(embeddedReport).toMatchSnapshot();
  });
});
