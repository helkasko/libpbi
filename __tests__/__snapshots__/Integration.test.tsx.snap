// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library User can render a report with embed token 1`] = `
<EmbeddedReport
  report={
    ReportWithEmbedToken {
      "baseUrl": "https://app.powerbi.com/reportEmbed",
      "embedConfig": Object {
        "tokenType": 1,
        "type": "report",
      },
      "reportConfig": Object {
        "accessToken": "token",
        "description": "description",
        "groupId": "groupId",
        "reportId": "reportId",
      },
    }
  }
>
  <div />
</EmbeddedReport>
`;

exports[`Library User can render a visual with embed token 1`] = `
<EmbeddedReport
  report={
    VisualWithEmbedToken {
      "baseUrl": "https://app.powerbi.com/reportEmbed",
      "embedConfig": Object {
        "tokenType": 1,
        "type": "visual",
      },
      "reportConfig": Object {
        "accessToken": "token",
        "description": "description",
        "groupId": "groupId",
        "pageName": "pageName",
        "reportId": "reportId",
        "visualName": "visualName",
      },
    }
  }
>
  <div />
</EmbeddedReport>
`;
